test: drivers: adc: Change the property names in the overlay

rruuaanng opened the pull request:
Change the property names in the bindings and DTS
to use hyphens(-) for separation instead of underscores(_).

anangl commented on tests/drivers/adc/adc_accuracy_test/boards/ek_ra4e2.overlay:
```diff
@@ -7,8 +7,8 @@
 		io-channels = <&adc0 0>;
-		reference_mv = <3300>;
-		expected_accuracy = <32>;
+		reference-mv = <3300>;
```
Please update also this line: https://github.com/zephyrproject-rtos/zephyr/blob/4068294b68337f9f36ba22acdfd042cdda39deb5/tests/drivers/adc/adc_accuracy_test/README.txt#L20,

rruuaanng replied:
Done.

rruuaanng committed:
tests: drivers: adc: Change the property names in the overlay

Unify property names in overlay, using hyphens(-)
instead of underscores (_) as separators.

Signed-off-by: James Roy <rruuaanng@outlook.com>
